core/drivers/imx_i2c.c

```diff
@@ -132,10 +144,28 @@ static void i2c_set_prescaler(uint8_t bid, uint32_t bps)
 
 static void i2c_set_bus_speed(uint8_t bid, int bps)
 {
-	/* Enable the clock */
-	io_write32(i2c_clk.base.va + CCM_CCGRx_SET(i2c_clk.i2c[bid]),
-		   CCM_CCGRx_ALWAYS_ON(0));
-
+	vaddr_t addr = i2c_clk.base.va;
+	uint32_t val = 0;
+
+#if defined(CFG_MX8MM)
+	addr += CCM_CCGRx_SET(i2c_clk.i2c[bid]);
+	val = CCM_CCGRx_ALWAYS_ON(0);
+#elif defined(CFG_MX6ULL)
+	switch (bid) {
+	case 0:
+		val = BM_CCM_CCGR2_I2C1_SERIAL;
```

Comment:
IMO, adding another macro would be better for consistency with the lines above.
```#define CCGR2_I2C_SERIAL(_x) BM_CCM_CCGR2_I2C##_x_SERIAL```
```val = CCGR2_I2C_SERIAL(bid)```

Reply:
ok. I think we missed this 3.11 window. I'll post something for the next release. closing it now.